### PR TITLE
Some clippy things

### PR DESCRIPTION
**Pull request by kjeremy**

### CODE REVIEW

File: crates/ra_assists/src/split_import.rs [hunk 1]
```diff
@@ -15,9 +15,7 @@ pub(crate) fn split_import(ctx: AssistCtx<impl HirDatabase>) -> Option<Assist> {
     let top_path = generate(Some(path), |it| it.parent_path()).last()?;
 
     let use_tree = top_path.syntax().ancestors().find_map(ast::UseTree::cast);
-    if use_tree.is_none() {
-        return None;
-    }
+    use_tree?;
```
Comment (matklad) [inline crates/ra_assists/src/split_import.rs]: The explicit version looks more readable to me. We can condense this to a single cryptic expression

```
top_path.syntax().ancestors().find_map(ast::UseTree::cast)?
```

but this seems unobvious to me.

File: crates/ra_lsp_server/src/server_world.rs [hunk 2]
```diff
@@ -73,7 +73,7 @@ impl ServerWorldState {
                     if let (Some(&from), Some(&to)) =
                         (sysroot_crates.get(&from), sysroot_crates.get(&to))
                     {
-                        if let Err(_) = crate_graph.add_dep(from, name.clone(), to) {
+                        if crate_graph.add_dep(from, name.clone(), to).is_err() {
```
Comment (matklad) [inline crates/ra_lsp_server/src/server_world.rs]: I prefer the former here: here we are doing a non-obvious thing, ignorring an error, and `_` makes that clearer.

File: crates/ra_lsp_server/src/server_world.rs [hunk 3]
```diff
@@ -121,7 +122,7 @@ impl ServerWorldState {
                         }
                     }
                     if let Some(std) = libstd {
-                        if let Err(_) = crate_graph.add_dep(from, "std".into(), std) {
+                        if crate_graph.add_dep(from, "std".into(), std).is_err() {
                             log::error!("cyclic dependency on std for {}", pkg.name(&ws.cargo))
                         }
```
Comment (matklad) [inline crates/ra_lsp_server/src/server_world.rs]: same here

File: crates/ra_lsp_server/src/server_world.rs [hunk 4]
```diff
@@ -134,7 +135,7 @@ impl ServerWorldState {
                 for dep in pkg.dependencies(&ws.cargo) {
                     if let Some(&to) = pkg_to_lib_crate.get(&dep.pkg) {
                         for &from in pkg_crates.get(&pkg).into_iter().flatten() {
-                            if let Err(_) = crate_graph.add_dep(from, dep.name.clone(), to) {
+                            if crate_graph.add_dep(from, dep.name.clone(), to).is_err() {
```
Comment (matklad) [inline crates/ra_lsp_server/src/server_world.rs]: and here

**Comment by kjeremy** [comment]
I reverted those bits and force pushed

**Pull request merged**